[ci] Python CI run 3.11 runs into timeout

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/fluss-rust/issues) and found nothing similar.


### Please describe the bug 🐞

https://github.com/apache/fluss-rust/actions/runs/24331087662/job/71036897707

```
=========================== short test summary info ============================
ERROR test/test_admin.py::test_error_database_not_exist - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_error_table_already_exist - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_error_table_not_exist - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_error_database_already_exist - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_create_database - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_get_server_nodes - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_kv_table.py::test_composite_primary_keys - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_create_table - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_error_table_not_partitioned - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_kv_table.py::test_partial_update - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_kv_table.py::test_upsert_delete_and_lookup - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_kv_table.py::test_partial_update_by_index - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_kv_table.py::test_partitioned_table_upsert_and_lookup - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_partition_apis - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_append_and_scan - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_kv_table.py::test_all_supported_datatypes - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_list_offsets - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_append_dict_rows - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_partitioned_table_append_scan - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_project - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_write_arrow - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_write_pandas - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_admin.py::test_fluss_error_response - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_partitioned_table_to_arrow - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_scan_records_indexing_and_slicing - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_async_iterator - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_async_iterator_break_no_leak - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_async_iterator_multiple_batches - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_batch_async_iterator_break_no_leak - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_batch_async_iterator - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_batch_async_iterator_multiple_batches - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_poll_batches - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_append_and_scan_with_array - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_append_rows_with_array - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_append_rows_with_nested_array - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_append_rows_with_invalid_array - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_sasl_auth.py::test_sasl_connect_with_valid_credentials - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_sasl_auth.py::test_sasl_connect_with_second_user - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_sasl_auth.py::test_sasl_connect_with_unknown_user - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_sasl_auth.py::test_sasl_client_to_plaintext_server - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_log_table.py::test_to_arrow_and_to_pandas - Failed: Timeout (>30.0s) from pytest-timeout.
ERROR test/test_sasl_auth.py::test_sasl_connect_with_wrong_password - Failed: Timeout (>30.0s) from pytest-timeout.
```

### Solution

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

From the error logs, it seems that there's a `_start_cluster` error with `gw1`:

```
# From your gw1 traceback
test/conftest.py:61: in _start_cluster
    result = subprocess.run(...)
...
    stdout, stderr = process.communicate(input, timeout=timeout)
E           Failed: Timeout (>30.0s) from pytest-timeout.
```

This potentially then lead to the other workers (`gw0`, `gw2`, and `gw3`) timing out. One simple test could be to add this to the `fluss-rust/bindings/python/pyproject.toml`:
```
[tool.pytest.ini_options]
timeout = 60
```

--- a/bindings/python/pyproject.toml
+++ b/bindings/python/pyproject.toml
@@ -96,7 +96,7 @@ known-first-party = ["fluss"]
 [tool.pytest.ini_options]
 asyncio_mode = "auto"
 asyncio_default_fixture_loop_scope = "function"
-timeout = 30
+timeout = 120
 
 [tool.mypy]
 python_version = "3.9"

--- a/bindings/python/pyproject.toml
+++ b/bindings/python/pyproject.toml
@@ -95,7 +95,7 @@ known-first-party = ["fluss"]
 
 [tool.pytest.ini_options]
 asyncio_mode = "auto"
-asyncio_default_fixture_loop_scope = "session"
+asyncio_default_fixture_loop_scope = "function"
 timeout = 30
 
 [tool.mypy]

--- a/bindings/python/test/conftest.py
+++ b/bindings/python/test/conftest.py
@@ -124,7 +124,7 @@ def fluss_cluster():
     yield (plaintext_addr, sasl_addr or plaintext_addr)
 
 
-@pytest_asyncio.fixture(scope="session")
+@pytest_asyncio.fixture
 async def connection(fluss_cluster):
     plaintext_addr, _sasl_addr = fluss_cluster
     conn = await _connect(plaintext_addr)
@@ -144,6 +144,6 @@ def plaintext_bootstrap_servers(fluss_cluster):
     return plaintext_addr
 
 
-@pytest_asyncio.fixture(scope="session")
+@pytest_asyncio.fixture
 async def admin(connection):
     return connection.get_admin()

--- a/bindings/python/test/conftest.py
+++ b/bindings/python/test/conftest.py
@@ -124,12 +124,16 @@ def fluss_cluster():
     yield (plaintext_addr, sasl_addr or plaintext_addr)
 
 
+_cached_connection = None
+
+
 @pytest_asyncio.fixture
 async def connection(fluss_cluster):
-    plaintext_addr, _sasl_addr = fluss_cluster
-    conn = await _connect(plaintext_addr)
-    yield conn
-    conn.close()
+    global _cached_connection
+    if _cached_connection is None:
+        plaintext_addr, _sasl_addr = fluss_cluster
+        _cached_connection = await _connect(plaintext_addr)
+    yield _cached_connection
 
 
 @pytest.fixture(scope="session")

